### PR TITLE
Update HtmlView to HtmlString instead of String

### PR DESCRIPTION
**Pull request by labkey-jeckels**
I'm not happy with ModuleHtmlView but didn't see an easy way to have access to a fully populated ViewContext where I need it

### CODE REVIEW

File: study/src/org/labkey/study/SpecimenManager.java [hunk 1]
```diff
@@ -3285,9 +3286,9 @@ public ExtendedSpecimenRequestView getExtendedSpecimenRequestView(ViewContext co
             {
                 ModuleHtmlView moduleView = ModuleHtmlView.get(module, path);
                 assert null != moduleView;
-                String html = moduleView.getHtml();
-                html = ModuleHtmlView.replaceTokens(html, context);
-                return ExtendedSpecimenRequestView.createView(html);
+                HtmlString html = moduleView.getHtml();
+                String s = ModuleHtmlView.replaceTokens(html.toString(), context);
+                return ExtendedSpecimenRequestView.createView(s);
```
Comment (labkey-matthewb) [inline study/src/org/labkey/study/SpecimenManager.java]: ExtendedSpecimenRequestView is "different"

**Comment by labkey-jeckels** [comment]
The premium build succeeded, which means that nobody else in the set of modules is calling HtmlView with null as the HTML, which is good for a number of reasons.

Just pushed a little more cleanup and converted a couple of places to DOM building.

File: experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java [hunk 2]
```diff
@@ -1528,50 +1540,42 @@ public void validateCommand(ToggleRunExperimentMembershipForm target, Errors err
...
+                        createTabSpacer(true))));
+    }
 
-        private void addTab(String text, ActionURL url, boolean selected, StringBuilder sb)
-        {
-            sb.append("<td class=\"labkey-tab" + (selected ? "-selected" : "") + "\" style=\"margin-bottom: 0px;\"><a href=\"" + url + "\">" + PageFlowUtil.filter(text) + "</a></td>");
-        }
+    private DOM.Renderable createTab(String text, ActionURL url, boolean selected)
+    {
+        return TD(cl("labkey-tab" + (selected ? "-selected" : "")),
```
Comment (ilikepies101) [inline experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java]: There's a cl() signature that takes in a boolean which you could use to mimic this behavior done with the selected ternary.
So the line would be:
`return TD(cl(selected, "labkey-tab-selected", "labkey-tab")),`

Comment (ilikepies101) [reply experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java]: Also, are we missing the style attribute here or is that no longer needed?

File: experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java [hunk 3]
```diff
@@ -4961,31 +4963,11 @@ public void setDataRegionSelectionKey(String key)
...
-        Iterator<ExpSampleSet> iter = sampleSets.iterator();
-        while (iter.hasNext())
-        {
-            ExpSampleSet sampleSet = iter.next();
-            if (!sampleSet.canImportMoreSamples())
-            {
-                iter.remove();
-            }
-        }
+        sampleSets.removeIf(sampleSet -> !sampleSet.canImportMoreSamples());
```
Comment (ilikepies101) [inline experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java]: This is cool.

File: experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java (hunk 2 above)
Comment (labkey-jeckels) [reply experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java]: The style setting the margin isn't needed, the page renders identically with or without it.

Just pushed a change that exposed the overloaded version of cl() on DOM itself, to make it more discoverable, and used it here.

Comment (ilikepies101) [reply experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java]: Nice! Didn't notice it wasn't static but thanks for adding that in.

**Pull request merged**